### PR TITLE
Add zone min and max volumes

### PR DESCRIPTION
**Pull request by Lohrer**

### CODE REVIEW

File: amplipi/rt.py [hunk 1]
```diff
@@ -523,15 +523,15 @@ def update_zone_vol(self, zone, vol):
 
       Args:
         zone: zone to adjust vol
-        vol: int in range[MIN_VOL, 0]
+        vol: int in range[MIN_VOL_DB, 0]
```
Comment (linknum23) [inline amplipi/rt.py:526]: Can this be an F-string?

File: amplipi/utils.py [hunk 2]
```diff
@@ -88,6 +89,14 @@ def clamp(xval, xmin, xmax):
   """ Clamp and value between min and max """
   return max(xmin, min(xval, xmax))
 
+def round_sf(x: float, sf: int = 3):
```
Comment (linknum23) [inline amplipi/utils.py:92]: used?

File: amplipi/models.py [hunk 3]
```diff
@@ -342,6 +384,7 @@ class GroupUpdate(BaseUpdate):
   zones: Optional[List[int]] = fields.Zones
   mute: Optional[bool] = fields.GroupMute
   vol_delta: Optional[int] = fields.GroupVolume
+  vol_delta_f: Optional[float] = fields.GroupVolumeF
```
Comment (linknum23) [inline amplipi/models.py:387]: lets call this `vol_f` with the intent of deprecating `vol_delta` to `vol`

File: tests/test_rest.py [hunk 4]
```diff
@@ -759,3 +772,211 @@ def on_service_state_change(zeroconf: Zeroconf, service_type: str, name: str, st
   # check advertisememts
   assert AMPLIPI_ZC_NAME in services_advertised
   assert services_advertised[AMPLIPI_ZC_NAME].port == 9898
+
```
Comment (linknum23) [inline tests/test_rest.py:775]: These tests are really valuable! I know they are pretty verbose but I don't know how much is gained by refactoring.

**Review by linknum23 — CHANGES_REQUESTED** [review]

File: amplipi/rt.py (hunk 1 above)
Comment (Lohrer) [reply amplipi/rt.py:526]: Not without a decorator, plus I'd argue that would mask how to properly use this method.

**Review by linknum23 — APPROVED** [review]

**Pull request merged**